chalice/awsclient.py

```diff
@@ -102,18 +122,60 @@ def _call_client_method_with_retries(self, method, kwargs):
         while True:
             try:
                 response = method(**kwargs)
-            except client.exceptions.InvalidParameterValueException:
+            except client.exceptions.InvalidParameterValueException as e:
                 # We're assuming that if we receive an
                 # InvalidParameterValueException, it's because
                 # the role we just created can't be used by
                 # Lambda so retry until it can be.
                 self._sleep(self.DELAY_TIME)
                 attempts += 1
-                if attempts >= self.LAMBDA_CREATE_ATTEMPTS:
+                message = e.response['Error'].get('Message', '')
+                retryable_message = (
+                    'The role defined for the function cannot be assumed '
+                    'by Lambda.'
+                )
+                if attempts >= self.LAMBDA_CREATE_ATTEMPTS or \
+                        retryable_message not in message:
                     raise
                 continue
             return response
 
+    def _raise_large_deployment_related_errors(self, error,
+                                               deployment_size):
+        # type: (Any, int) -> None
+        error_message = str(error)
+        should_raise = False
+        additional_details = None
+        if (isinstance(error, RequestsConnectionError) and
+                deployment_size > self.MAX_LAMBDA_DEPLOYMENT_SIZE):
+            # When the zip deployment package is too large and Lambda
+            # aborts the connection as chalice is still sending it
+            # data
+            additional_details = (
+                'This is likely because the deployment package is %s. '
```

Comment:
Lets move this up the stack and have stuff in awsclient just collect all the raw data, deployment package size, original exceptions, etc.   We'd then have the Exception object assemble the suggestions and pretty print the error messages to the user.